test/components/preprocessors/test_recursive_splitter.py

```diff
@@ -0,0 +1,258 @@
+
+    doc = Document(content=text)
+    doc_chunks = splitter.run([doc])
+    doc_chunks = doc_chunks["documents"]
+
+    assert len(doc_chunks) == 4
+    assert (
+        doc_chunks[0].meta["original_id"]
+        == doc_chunks[1].meta["original_id"]
+        == doc_chunks[2].meta["original_id"]
+        == doc_chunks[3].meta["original_id"]
+        == doc.id
+    )
+    assert doc_chunks[0].content == "Artificial intelligence (AI) - Introduction\n\n"
+    assert (
+        doc_chunks[1].content
+        == "AI, in its broadest sense, is intelligence exhibited by machines, particularly computer systems.\n"
+    )
+    assert doc_chunks[2].content == "AI technology is widely used throughout industry, government, and science."
+    assert (
+        doc_chunks[3].content
+        == " Some high-profile applications include advanced web search engines; recommendation systems; interacting via human speech; autonomous vehicles; generative and creative tools; and superhuman play and analysis in strategy games."
+    )
+
+
+def test_recursive_chunker_split_document_with_overlap():
+    splitter = RecursiveDocumentSplitter(split_length=20, split_overlap=11, separators=[".", " "], keep_separator=True)
+    text = """A simple sentence1. A bright sentence2. A clever sentence3. A joyful sentence4"""
+
+    doc = Document(content=text)
+    doc_chunks = splitter.run([doc])
+    doc_chunks = doc_chunks["documents"]
+
+    assert len(doc_chunks) == 4
+    assert (
+        doc_chunks[0].meta["original_id"]
+        == doc_chunks[1].meta["original_id"]
+        == doc_chunks[2].meta["original_id"]
+        == doc_chunks[3].meta["original_id"]
+        == doc.id
+    )
+
+    assert doc_chunks[0].content == "A simple sentence1."
+    assert doc_chunks[0].meta["split_id"] == 0
+    assert doc_chunks[0].meta["split_idx_start"] == 0
+    assert doc_chunks[0].meta["_split_overlap"] == [{"doc_id": doc_chunks[1].id, "range": (0, 11)}]
+
+    assert doc_chunks[1].content == " sentence1. A bright sentence2."
+    assert doc_chunks[1].meta["split_id"] == 1
+    assert doc_chunks[1].meta["split_idx_start"] == 8
+    assert doc_chunks[1].meta["_split_overlap"] == [
+        {"doc_id": doc_chunks[0].id, "range": (8, 19)},
+        {"doc_id": doc_chunks[2].id, "range": (0, 11)},
+    ]
+
+    assert doc_chunks[2].content == " sentence2. A clever sentence3."
+    assert doc_chunks[2].meta["split_id"] == 2
+    assert doc_chunks[2].meta["split_idx_start"] == 28
+    assert doc_chunks[2].meta["_split_overlap"] == [
+        {"doc_id": doc_chunks[1].id, "range": (20, 31)},
+        {"doc_id": doc_chunks[3].id, "range": (0, 11)},
+    ]
+
+    assert doc_chunks[3].content == " sentence3. A joyful sentence4"
+    assert doc_chunks[3].meta["split_id"] == 3
+    assert doc_chunks[3].meta["split_idx_start"] == 48
+    assert doc_chunks[3].meta["_split_overlap"] == [{"doc_id": doc_chunks[2].id, "range": (20, 31)}]
+
+
+def test_recursive_splitter_generate_pages():
```

Comment:
In addition to this test could we add tests like this:
- https://github.com/deepset-ai/haystack/blob/04fc187bc44c39fcd5d8c29ce7f27f91667065ce/test/components/preprocessors/test_nltk_document_splitter.py#L311
to make sure page counting still works with empty pages and when using split_overlap? 
- Also could we test if page counting works with a variety of different separators? So separate test for `\f`, `\n`, and `"sentence"` in addition to this one which you have for splitting by word

Reply:
added tests for this one as well